### PR TITLE
Backend-APIs-SignUp-Login

### PR DESCRIPTION
**Pull request by ManpreetBhuller**
User.js- User Model
user.js- register and login API for user
middleware/auth.js- for token

### CODE REVIEW

**Comment by tonyc856** [comment]
I think there's a bit of code formatting discrepancy (not a huge deal but it's just important to have a consistent code style when possible). Were you able to install the Prettier plugin? Try running it again and see if it auto-reformatted the code correctly.

Other than that, it's looking good! Great work

File: server/middleware/auth.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+module.exports = function(req, res, next){
+
+    const token = req.header('x-auth-token');
+    if(!token){
+        return res.status(401).json({msg:'No token, authorization denied'});
+
+    }
+    try{
+        const decoded = jwt.verify(token, config.get('jwtSecret'));
+        req.user = decoded.user;
```
Comment (bonnieli) [inline server/middleware/auth.js:13]: is this file being used? 

why is it changing `req` and added the user to it?

File: server/routes/api/user.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        });
+    }catch(err){
+        res.status(500).send('Server error');
+    }
+
+});
+
+//api for login
+router.post('/login', async (req, res, next) => {
+    const {email, password} = req,body;
```
Comment (reny1cao) [inline server/routes/api/user.js]: There is a small bug

Comment (bonnieli) [reply server/routes/api/user.js]: to expand on Renyi's comment, I believe it should be `req.body`

Comment (reny1cao) [reply server/routes/api/user.js]: Yes. Sorry I was a bit vague here.

File: client/src/index.js [hunk 3]
```diff
@@ -2,11 +2,11 @@ import React from 'react';
...
-import * as serviceWorker from './serviceWorker';
+//import * as serviceWorker from './serviceWorker';
 
 ReactDOM.render(<App />, document.getElementById('root'));
 
 // If you want your app to work offline and load faster, you can change
 // unregister() to register() below. Note this comes with some pitfalls.
 // Learn more about service workers: https://bit.ly/CRA-PWA
-serviceWorker.unregister();
+//serviceWorker.unregister();
```
Comment (tonyc856) [inline client/src/index.js:12]: I was just wondering why this is being commented out?

**Comment by bonnieli** [comment]
update branch to resolve merge conflicts

File: client/src/index.js (hunk 3 above)
Comment (ManpreetBhuller) [reply client/src/index.js:12]: Oh my bad it was by mistake.

**Pull request merged**